watching for messages now includes ones own from other devices

malgorithms opened the pull request:
@nathunsmitty, small changes here although lots diffs in chat.test.js because the bot `alice` is now `alice1` and `alice2`.

the key change here is that watching for new messages now by default includes your own message sent from other devices. I decided to resolve the issue (#65) in the simplest imaginable way: just let the user look at the data themselves if they want to filter these out. 

so, watching for new messages:
- includes messages from others
- includes messages from your other devices 
- does not include messages from the same device that's running the bot

also note I call your `formatAPIObjectOutput` on the watched messages, which we weren't doing.

nathunsmitty commented on __tests__/chat.test.js:
```diff
@@ -219,46 +222,67 @@ describe('Chat Methods', () => {
     })
+
+    it("Doesn't pick up its own messages from the same device", async () => {
+      const messageText = 'Ever thus to deadbeats, Lebowski'
```
😆

nathunsmitty commented on lib/chat-client/index.js:
```diff
@@ -196,14 +198,17 @@ class Chat extends ClientBase {
     const onLine = (line: string) => {
       try {
-        const messageObject: MessageNotification = JSON.parse(line)
+        const messageObject: MessageNotification = formatAPIObjectOutput(JSON.parse(line))
```
glad we changed this! thanks for catching that 🎉

nathunsmitty commented on __tests__/chat.test.js:
```diff
@@ -219,46 +222,67 @@ describe('Chat Methods', () => {
-                const reply = {body: `${CONVO_CODE} ${num + 1}`}
-                await bot.chat.send(message.channel, reply)
-              }
+  const testTwoBotsCounting = async (bot1, bot2) => {
```
I think it makes more sense to put this in the describe block for `watchAllChannelsForNewMessages`

malgorithms replied:
cool yeah, moving it in there.

malgorithms commented:
ok @nathunsmitty  - ran `yarn docs`. also changed the test. I think it was just timing out on your machine, in the 10 second internal timeout that test had. I just removed the 10 seconds from it, which seemed unnecessary since jest is doing a timeout on the whole thing anyway. 

interestingly, I couldn't do this because it wouldn't pass eslint:

```
      while (highestReached < stopAt) {
        await timeout(100)
      }
```

because the linter assumes it's impossible for that to ever be satisfied.  I guess it doesn't understand awaits well enough...anyway, I changed it to a slight variation.

all tests pass for me now with the exception of deinit which fails once in a while, but as you said that's an issue in master.

also moved the testTwoBotsCounting into the describe block as you requested.
